cores/esp8266/umm_malloc/umm_malloc.cpp

```diff
@@ -1462,26 +1586,31 @@ static void *_umm_malloc( size_t size ) {
       UMM_PFREE( UMM_NFREE(cf) ) = cf + blocks;
       UMM_NFREE( cf + blocks ) = UMM_NFREE(cf);
     }
+
+    /* Update dynamic Free Block count */
+    ummHeapFreeBlocks -= blocks;
+
   } else {
     /* Out of memory */
 
     DBG_LOG_DEBUG(  "Can't allocate %5d blocks\n", blocks );
 
     /* Release the critical section... */
-    UMM_CRITICAL_EXIT();
+    UMM_CRITICAL_EXIT(id_malloc);
```

Comment:
Same comment as above, maybe move back before the debug.  In this case, it's probably a fatal app error anyway, so not a big deal.

Reply:
This may be a style issue. I moved it before the `Out of memory` comment. So that the debug print is closest to the comment.